web/views/metrics.erb

```diff
@@ -1,50 +1,58 @@
+<script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/Chart.js/3.9.1/chart.min.js"></script>
+<script type="text/javascript" src="<%= root_path %>javascripts/metrics.js"></script>
 
-<h1><%= t('Metrics') %></h1>
+<h3><%= t('Metrics') %></h3>
 
-<h3>Top Jobs by Processed Count</h3>
-<% top = @resultset[:top_classes] %>
+<%
+  limit = 10
+  job_results = @query_result.job_results.sort_by { |(kls, jr)| jr.totals["ms"] }.reverse.first(limit).to_h
+  # Colors taken from https://www.chartjs.org/docs/latest/samples/utils.html
+  colors = [
+    "#4dc9f6",
+    "#f67019",
+    "#f53794",
+    "#537bc4",
+    "#acc236",
+    "#166a8f",
+    "#00a950",
+    "#58595b",
+    "#8549ba",
+    "#991b1b",
+  ]
+%>
 
-<% topp = top["p"]&.first(10) %>
-<div class="table_container">
-  <table class="table table-bordered table-striped table-hover">
-    <tbody>
-      <tr>
-        <th><%= t('Name') %></th>
-        <th><%= t('Processed') %></th>
-        <th><%= t('ExecutionTime') %></th>
-      </tr>
-      <% if topp %>
-        <% topp.each do |kls, val| %>
-          <tr>
-            <td><code><a href="<%= root_path %>metrics/<%= kls %>"><%= kls %></a></code></td>
-            <td><%= val %></td>
-            <td><%= top.dig("ms", kls) %></td>
-          </tr>
-        <% end %>
-      <% else %>
-          <tr><td colspan=3><%= t("NoDataFound") %></td></tr>
-      <% end %>
-    </tbody>
-  </table>
-</div>
+<canvas
+  id="metrics-chart"
+  data-labels="<%= h Sidekiq.dump_json(@query_result.buckets) %>"
+  data-series="<%= h Sidekiq.dump_json(job_results.map { |(kls, jr)| [kls, jr.dig("series", "ms")] }.to_h) %>"
+  data-colors="<%= h Sidekiq.dump_json(colors) %>"
+></canvas>
 
-<h3>Top Jobs by Execution Time</h3>
+<h3>Most Time-Consuming Jobs</h3>
```

Comment:
Renamed this in case we want to make this table sortable. It will also list the top X most time-consuming jobs, but they might not be _sorted by_ execution time (although they are right now).